### PR TITLE
Change TansactionCommitment input from []*TransactionReceipt to []Transaction 

### PR DESCRIPTION
**Pull request by IronGauntlets**
Transaction commitment is calculated using transactions instead of transaction receipts.

## Changes:

See commits

## Types of changes

_Leave only items that describe your changes, remove the rest. Remove this line too._

- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Documentation Update
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no api changes)
- 
## Testing

**Requires testing**: No

**Did you write tests??**: Yes

### CODE REVIEW

File: core/transaction.go [hunk 1]
```diff
@@ -220,36 +279,33 @@ func declareTransactionHash(d *DeclareTransaction, network utils.Network) (*felt
...
+func TransactionCommitment(transactions []Transaction) (*felt.Felt, error) {
 	var transactionCommitment *felt.Felt
 	return transactionCommitment, trie.RunOnTempTrie(commitmentTrieHeight, func(trie *trie.Trie) error {
-		for i, receipt := range receipts {
-			signaturesHash := crypto.PedersenArray()
-			if receipt.Type == Invoke {
-				signaturesHash = crypto.PedersenArray(receipt.Signatures...)
+		for i, transaction := range transactions {
+			signatureHash := crypto.PedersenArray()
+			if _, ok := transaction.(*InvokeTransaction); ok {
```
Comment (omerfirmak) [inline core/transaction.go:303]: this if is not needed anymore, right? signature() will return an empty array for non-invoke txs.

**Review by omerfirmak — APPROVED** [review]

File: core/transaction.go (hunk 1 above)
Comment (jelilat) [reply core/transaction.go:303]: Some other transaction types actually have a Signature (e.g. Declare)

**Review by jelilat — APPROVED** [review]

File: core/transaction.go (hunk 1 above)
Comment (omerfirmak) [reply core/transaction.go:303]: then if must go

File: core/transaction.go [hunk 2]
```diff
@@ -224,18 +279,69 @@ func (d *DeclareTransaction) Hash(network utils.Network) (*felt.Felt, error) {
...
+const commitmentTrieHeight uint = 64
+
+// TransactionCommitment is the root of a height 64 binary Merkle Patricia tree of the
+// transaction hashes and signatures in a block.
+func TransactionCommitment(transactions []Transaction) (*felt.Felt, error) {
+	var transactionCommitment *felt.Felt
+	return transactionCommitment, trie.RunOnTempTrie(commitmentTrieHeight, func(trie *trie.Trie) error {
+		for i, transaction := range transactions {
+			signatureHash := crypto.PedersenArray()
+			if _, ok := transaction.(*InvokeTransaction); ok {
```
Comment (omerfirmak) [inline core/transaction.go:303]: if should be removed

**Review by omerfirmak — CHANGES_REQUESTED** [review]

File: core/transaction.go (hunk 2 above)
Comment (jelilat) [reply core/transaction.go:303]: Not really. Although there are other transactions with Signatures, we only use that of `InvokeTransaction` to calculate commitments

Comment (omerfirmak) [reply core/transaction.go:303]: my bad

**Review by omerfirmak — APPROVED** [review]

**Pull request merged**